src/org/sipdroid/sipua/PublishAgent.java

```diff
@@ -109,16 +98,21 @@ public void publish(BasicStatus status, int expireTime, String note) {
 				e.printStackTrace();
 			}
 			String entity = "sip:" + user_profile.from_url;
-			String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
-							"<presence xmlns=\"urn:ietf:params:xml:ns:pidf\"" +
-							" entity=\"" + entity + "\">" +
-							"<tuple id=\"" + tupleId + "\">" +
-							"<status>" +
-							"<basic>" + status.pidf() + "</basic>" +
-							"<note>" + note + "</note>" +
-							"</status>" +
-							"</tuple>" +
-							"</presence>";
+			String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>" +
+					"<presence xmlns=\"urn:ietf:params:xml:ns:pidf\"\n" +
+					"xmlns:dm=\"urn:ietf:params:xml:ns:pidf:data-model\"\n" +
+					"xmlns:rpid=\"urn:ietf:params:xml:ns:pidf:rpid\"" +
+					" entity=\"" + entity + "\">" +
+					" <dm:person id=\"" + "ID-" + tupleId + ">" +

```

Comment:
The person ID and the tuple id cannot be the same. Maybe prefix them with PID- and TID- respectively or the document won't validate.